README.md

```diff
@@ -42,18 +42,15 @@ fastify.register(require('fastify-accepts-serializer'), {
 
 // Per-router serializers
 const config = {
-  serializer: {
-    serializers: [
-      {
-        regex: /^application\/x-protobuf$/,
-        serializer: body => AwesomeMessage.encode(AwesomeMessage.create(body)).finish()
-      }
-    ]
-  }
+  type: 'application/x-protobug',
+  serializer: body => AwesomeMessage.encode(AwesomeMessage.create(body)).finish()
 }
 
-fastify.get('/request', { config }, function (req, reply) {
-  reply.send({pippo: 'pluto'})
+fastify.get('/request', function (req, reply) {
+  reply
+    .type(config.type)
+    .serializer(config.serializer)
+    .send({pippo: 'pluto'})
 })
```

Comment:
I do not understand the intention of this change. In the original example, the serializer is being added to a configuration object that conforms to a configuration expected by this module's `SerializerManager` object. That configuration is then attached to an individual route via Fastify's standard per-route configuration options. The result being, that configuration object can be used with multiple routes.

Your change keeps that reusable configuration object, but then does not pass it to the route. Instead, your example relies on that configuration object always being within scope of the handler function. This is a significant performance downgrade in implementations.

What are you trying to solve with this documentation change?

Reply:
passing to the config is not work because the related code is removed in #4.
so, we need to use the fastify reply ```type``` and ```serializer``` for route-level serializer.
the change of this document is remove the invalid example and put back the invalid one.

Reply:
actually, here are two way to solve issue #33
1. update the document 
2. write back the route level serializer function

however, I can see that all the ppl just skip the reply in https://github.com/fastify/fastify-accepts-serializer/pull/4#issuecomment-439709250 .
I am willing to add the function back.

Reply:
I looked through the current code of this plugin prior to my last comment. I think the only change that needs to be made is to make `serializers` a root property of the configuration object instead of a child of `serializer`.

Reply:
I did not know this PR is attempting to resolve #32. In future PRs, please reference any issues they attempt to resolve. It helps inform the review process.

Reply:
Yeah, reading #32 confirms what I deduced from reading the code. The example just needs to be changed from:

```js
const config = {
  serializer: {
    serializers: []
  }
}
```

to

```js
const config = {
  serializers: []
}
```

Reply:
If my understanding is correct the example will becomes the below one. and if I am correct, I have tested it wont work.
```javascript
const config = {
  serializers: [
    {
      regex: /^application\/x-protobuf$/,
      serializer: body => AwesomeMessage.encode(AwesomeMessage.create(body)).finish()
    }
  ]
}

fastify.get('/request', { config }, function (req, reply) {
  reply.send({pippo: 'pluto'})
})
```

Reply:
Yeah, that's what I'm suggesting. If that doesn't work then it seems like a bug that needs to be fixed, because my reading of the code suggests that it should work.

Reply:
The code never access ```reply.context.config```, so it will never use the config passed through the route options. If you want to add the functionality back in place, we should do a lot of work on the current code, especially about the cache. I will assume it is another PR.

Reply:
Yeah, I think it is better to fix the bug than work around it. Are you willing to work on that?

Reply:
I have updated the PR and add the test for few use case, you can check if it meet your assumption.